packit/upstream.py

```diff
@@ -414,6 +415,17 @@ def set_spec_version(self, version: str, changelog_entry: str):
             logger.error(f"rebase-helper failed to change the spec file: {ex!r}")
             raise PackitException("rebase-helper didn't do the job")
 
+    def get_archive_extension(self, archive_basename: str) -> str:
+        """
+        Obtains archive extension from SpecFile based on basename of the archive.
+        Defaults to .tar.gz if no Source corresponds to the basename.
+        """
+        for source in self.specfile.get_sources():
+            base = os.path.basename(source)
```

Comment:
I'm just thinking if it's better to iterate over the common tarball extensions here and just do if `source.endswith(...`

Reply:
Good point, will definitely make things a lot easier

Reply:
Now that I am looking at it a bit more closely, it could have some negative effects. If we just checked for the end and compared it to the common extensions, the base name of the archive wouldn't be taken into account, hence an incorrect extension could be returned. Consider this minimal example:
```
Name: foo
Source0: bar-%{version}.tar.gz
Source1: %{name}-%{version}.tar.bz2
```
`.tar.gz` would be returned instead of `.tar.bz2` (since that is the first source with matching extension). Examples like these are definitely not common (Source0 is usually the main archive) but still possible. What's your take on this?

Reply:
wow, very nice thinking!

okay, so let's do it your way, that should be more fool-proof